<a href="https://colab.research.google.com/github/victoraccete/LED_tasks/blob/main/Parte_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Parte 3

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

Devido ao tamanho do dataset e limitações de hardware, optei por usar o Colab em conjunto com uma montagem do Drive, onde subi o dataset. Optei por usar o Colab mesmo para ter acesso a mais recursos computacionais. 

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [10]:
older_data_path = '/content/drive/MyDrive/Dataset LED task/dados_AL1806.csv'
newer_data_path = '/content/drive/MyDrive/Dataset LED task/part-00000-8031ffb6-c8db-4d3a-a2e9-40384eaae292.c000.csv'
older_df = pd.read_csv(older_data_path, sep=';')
newer_df = pd.read_csv(newer_data_path, sep=';')

print(older_df.shape)
older_df.head()

(1156025, 34)


document_id  ... id_sistema_origem
0  ed76f33f-8a8b-4b16-a948-80abb05d0524-i0b0  ...               NaN
1  2a170500-4972-43a6-bb66-e3ada213e7ea-i0b0  ...               NaN
2  224187bc-7e01-4db5-8a0d-228ee708c69c-i0b0  ...               NaN
3  4cf5cdef-b88b-4713-b371-902af3da21cd-i0b0  ...               NaN
4  5f758e5e-25b7-4ad4-a479-c88bfb2a3496-i0b0  ...               NaN

[5 rows x 34 columns]

In [11]:
print(newer_df.shape)
newer_df.head()

(1172381, 34)


document_id  ... id_sistema_origem
0  7e93a8b5-ef3a-4541-a356-3fa29b440572-i0b0  ...               NaN
1  600389af-cbd6-4cc5-8cdd-4911cdac338c-i0b0  ...               NaN
2  7e9d2710-f993-4631-be42-5abc197df0a5-i0b0  ...               NaN
3  60430e15-9afb-4f8a-ac8d-dd88dbf3d7b1-i0b0  ...               NaN
4  4ceb44a2-e06a-4c69-a3a0-19dabc00381c-i0b0  ...               NaN

[5 rows x 34 columns]

In [15]:
similar_records = pd.merge(older_df, newer_df, how='inner')
print(similar_records.shape)
similar_records.head()

(1157837, 34)


document_id  ... id_sistema_origem
0  ed76f33f-8a8b-4b16-a948-80abb05d0524-i0b0  ...               NaN
1  2a170500-4972-43a6-bb66-e3ada213e7ea-i0b0  ...               NaN
2  224187bc-7e01-4db5-8a0d-228ee708c69c-i0b0  ...               NaN
3  4cf5cdef-b88b-4713-b371-902af3da21cd-i0b0  ...               NaN
4  5f758e5e-25b7-4ad4-a479-c88bfb2a3496-i0b0  ...               NaN

[5 rows x 34 columns]

Vemos que há um número significativo de registros iguais nas duas bases: 1157837. 